### DISCUSSION TITLE
HyperExpress: Uncaught Exception Occured

### DISCUSSION THREAD

**Discussion by timfemey** [kartikk221/hyper-express#87] [Q&A] [answered]
I get an error when starting my server  saying `HyperExpress: Uncaught Exception Occured`

Please can someone explain what causes this error and possible solutions

Thanks

**Comment by kartikk221** [comment]
Hi @timfemey can you provide the full error stack trace you see in your console when running your application?

**Comment by timfemey** [comment, marked as answer]
> Hi @timfemey can you provide the full error stack trace you see in your console when running your application?

There is no error in the console and it just returns a response with status of 500 saying HyperExpress: Uncaught Exception Occured.

I also tried catching the error with 
`process.on('uncaughtException',err=>{
 console.log(err)
}) ` 
but no errors were sent as standard output on the console

**Reply by kartikk221** [reply]
@timfemey I see, that is occuring because you currently do not have an error handler bound to HyperExpress.
Bind an error handler with the below snippet:
```js
webserver.set_error_handler((request, response, error) => {
    // Log the error
    console.log(error);

    // Send a response to the client
    response.status(500).send('Error Occured');
});
```

**Reply by timfemey** [reply]
I have added the error handler, i get this error 

`Error: HyperExpress: Double middleware execution detected! You have a bug where one of your middlewares is calling both the next() callback and also resolving from a Promise/async middleware. You must only use one of these not both.`

**Reply by kartikk221** [reply]
Check all of your middlewares that you are using to see If you have a middleware that is `async` and you are calling the `next()` method.
For example, the following snippet would create a double middleware execution:
```js
webserver.use(async (request, response, next) => {
    // You should not be calling "next()" in an async callback
    // async callback type middlewares are automatically awaited by HyperExpress 
    next();
});
```

**Reply by timfemey** [reply]
Thanks for the help, It works now 👍
